src/components/Common/UI/Button/ButtonIcon.tsx

```diff
@@ -0,0 +1,40 @@
+import { Button as AriaButton } from 'react-aria-components'
+import { type ButtonIconProps } from './ButtonTypes'
+import styles from './Button.module.scss'
+
+export function ButtonIcon({
+  isError = false,
+  isLoading = false,
+  isDisabled = false,
+  ref,
+  className,
+  children,
+  onBlur,
+  onFocus,
+  onClick,
+  ...props
+}: ButtonIconProps) {
+  const handlePress = onClick
+    ? () => {
+        onClick({} as React.MouseEvent<HTMLButtonElement>)
+      }
+    : undefined
+
+  return (
+    <span className={styles.root}>
+      <AriaButton
+        {...props}
+        ref={ref}
+        onBlur={onBlur}
+        onFocus={onFocus}
+        isDisabled={isDisabled || isLoading}
+        data-variant="icon"
+        data-loading={isLoading || undefined}
+        data-error={isError || undefined}
+        onPress={handlePress}
+      >
+        {children}
+      </AriaButton>
+    </span>
```

Comment:
Could this just be updated to use `Button.tsx`? then it wouldn't have to import the module styles from Button or duplicate this markup and the react aria usage stays in a single component